firestore/FirestoreSwiftUIExample/ViewModels/RestaurantViewModel.swift

```diff
@@ -35,6 +35,26 @@ class RestaurantViewModel: ObservableObject {
     unsubscribe()
   }
 
+  func add(review: Review) {
+    do {
+      _ = try restaurant.reference!.collection("ratings").addDocument(from: review)
+    } catch {
+      fatalError("Unable to add review: \(error.localizedDescription).")
+    }
+
+    let newAverage = (Float(restaurant.ratingCount) * restaurant.averageRating
+                        + Float(review.rating)) / Float(restaurant.ratingCount + 1)
+
+    restaurant.reference!.setData([
+      "numRatings": restaurant.ratingCount + 1,
+      "avgRating": newAverage
+    ], merge: true) { err in
+      if let err = err {
+        print("Error writing document: \(err)")
+      }
+    }
```

Comment:
The calculation of the average should be done in a [transaction](https://firebase.google.com/docs/firestore/manage-data/transactions). Looks like RestaurantViewModel may already have the necessary machinery for adding reviews in transactions.